src/strategies/uniswap/index.ts

```diff
@@ -0,0 +1,46 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+import { abi as uniAbi } from './UniV2.json';
+import { abi as tokenAbi } from '../erc20-balance-of/TestToken.json';
+
+//options.address is the Uni-V2 token address
+//options.tokenAddress is the token we are extracting
+export async function strategy(network, provider, addresses, options, snapshot) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const res = await Promise.all([
+    multicall(
+      network,
+      provider,
+      uniAbi,
+      [[options.address, 'totalSupply', []]],
+      { blockTag }
+    ),
+    multicall(
+      network,
+      provider,
+      tokenAbi,
+      [[options.tokenAddress, 'balanceOf', [options.address]]],
```

Comment:
I'm not sure to understand why you request `balanceOf` for both the pool token and the token itself. Can you explain?

Reply:
For example in the BUILD-ETH uniswap pool, I need to know how many BUILD tokens there are in 1 UNI-V2 token. So I check BUILD's balanceOf the UNI-V2 token address, and get a figure e.g. 10,000 BUILD in total in the contract. I divide that by the UNI-V2 totalSupply to get e.g. 1.5 BUILD per UNI-V2. Then I get UNI-V2's balanceOf the user address, to see how many UNI-V2 tokens the user has. So, if a user has 10 UNI-V2 tokens in the above example, they have 15 BUILD. I don't think it's possible to do this with less than 3 method calls, but of course I could be missing something.